pybedlite/overlap_detector.py

```diff
@@ -125,6 +126,66 @@ def from_bedrecord(cls: Type["Interval"], record: BedRecord) -> "Interval":
             name=record.name,
         )
 
+    @classmethod
+    def from_ucsc_position(
+        cls: Type["Interval"],
+        position: str,
+        name: str | None = None,
+    ) -> "Interval":
+        """
+        Construct an `Interval` from a UCSC "position"-formatted string.
+
+        The "Position" format (referring to the "1-start, fully-closed" system as coordinates are
+        "positioned" in the browser)
+            * Written as: chr1:127140001-127140001
+            * The location may optionally be followed by a parenthetically enclosed strand, e.g.
+              chr1:127140001-127140001(+).
+            * No spaces.
+            * Includes punctuation: a colon after the chromosome, and a dash between the start and
+              end coordinates.
+            * When in this format, the assumption is that the coordinate is **1-start,
+              fully-closed.**
+        https://genome-blog.gi.ucsc.edu/blog/2016/12/12/the-ucsc-genome-browser-coordinate-counting-systems/  # noqa: E501
+
+        Note that when the string does not have a specified strand, the `Interval`'s negative
+        attribute is set to False. This mimics the behavior of `OverlapDetector.from_bed()` when
+        reading a record that does not have a specified strand.
+
+        Args:
+            position: The UCSC "position"-formatted string.
+            name: An optional name for the interval.
+
+        Returns:
+            An `Interval` corresponding to the same region specified in the string.
+            Note that the `Interval` is **zero-based open-ended**.
+
+        Raises:
+            ValueError: If the string is not a valid UCSC position-formatted string.
+        """
+
+        # First, check to see if the strand is specified, and remove it from the string.
+        strand_re = re.compile(r".*\((\+|-)\)$")
+        strand_match = strand_re.match(position)
+
+        if strand_match is not None:
+            negative = strand_match.group(1) == "-"
+            position = position[:-3]
+        else:
+            negative = False
+
+        # Then parse the location
+        position_re = re.compile(r"^(chr(\d+|X|Y|M|MT)(?:_[A-Za-z0-9]+_alt)?):(\d+)-(\d+)$")
```

Comment:
1. ditto about pre-compiling
2. I am confused by this regex and why it has so much assumed about the contig name?  Have you thought about this deeply for non-human or alternate/decoy/custom contig names?  For example `chrUn_JTFH01001499v1_decoy` doesn't parse and neither does `HLA-DRB1*15:01:01:02`.

Reply:
Good points both.

Should we accept any string of characters followed by the start/end (i.e. `(.*):(\d+)-(\d+)$`) ? Or are there any constraints you think we should impose on the contig name?

Reply:
Lets use the regex in the SAM spec?  Alternatively, why not just split on the last colon?

Reply:
If you're fine splitting on the last colon, this is functionally equivalent and imo slightly more elegant since every element (`refname`, `start`, and `end`) is represented in a matched `group`

Reply:
If you have a strong preference I'll put in the SAM spec regex. It wouldn't be my preference as it's somewhat gnarly and I don't think python supports defining custom character classes to simplify, the way `[:rname:]` is used in the spec

```
[0-9A-Za-z!#$%&+./:;?@^_|~-][0-9A-Za-z!#$%&*+./:;=?@^_|~-]*
```

Reply:
@nh13 bump 🙂